lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -141,56 +184,80 @@ private void processUsersToFindLOCAndUpdateDb(List<String> userIds)
                                 String.format("User (%s) VCs not correlated.", userId)));
                 continue;
             }
-            boolean migrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+            boolean anyVCsMigrated = tacticalVcs.stream().anyMatch(vc -> vc.getMigrated() != null);
+            boolean allVCsMigrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+
+            if (anyVCsMigrated && !allVCsMigrated) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Not all VCs are migrated for this user (%s).", userId)));
+            }
             var votAttained =
                     reportUserIdentityService.getStrongestAttainedVotForCredentials(tacticalVcs);
 
             reportUserIdentityItems.add(
                     new ReportUserIdentityItem(
                             userId,
                             votAttained.orElse(Vot.P0).name(),
+                            tacticalVcs.size(),
                             reportUserIdentityService.getIdentityConstituent(tacticalVcs),
-                            migrated));
+                            allVCsMigrated));
         }
         LOGGER.info(LogHelper.buildLogMessage("Updating processed user's identity."));
         reportUserIdentityDataStore.createOrUpdate(reportUserIdentityItems);
     }
 
-    private ReportProcessingResult buildReportProcessingResult() {
+    private ReportProcessingResult.ReportProcessingResultBuilder buildReportProcessingResult(
+            ReportProcessingRequest reportProcessingRequest,
+            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult) {
         LOGGER.info(LogHelper.buildLogMessage("Building report processing summary result."));
+        Map<String, AttributeValue> exclusiveStartKey = null;
+        if (reportProcessingRequest.userIdentitylastEvaluatedKey() != null) {
+            exclusiveStartKey = reportProcessingRequest.userIdentitylastEvaluatedKey();
+        }
+        List<ReportUserIdentityItem> userIdentities = new ArrayList<>();
```

Comment:
This is still aggregating the entire table into a single object. We need to process the aggregate metrics on a page-by-page basis, something like:
```
var totalP2 = 0L;
var totalP1 = 0L;
// etc.
do {
    var tableScanResult =
            reportUserIdentityDataStore.getItems(exclusiveStartKey, "identity", "migrated");
    for (var item : tableScanResult.items()) {
        switch (item.getIdentity()) {
            case "P2" -> totalP2++;
            case "P1" -> totalP1++;
        }
        // etc.
    }
    exclusiveStartKey = tableScanResult.lastEvaluatedKey();
} while (exclusiveStartKey != null);
```
That way each scan result can be garbage-collected after it's processed, and we only store the aggregate statistics at the end.

Reply:
Handle it in processUsersToFindLOCAndUpdateDb()